sdk/core/System.ClientModel/src/Pipeline/ClientLoggingPolicy.cs

```diff
@@ -0,0 +1,363 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A <see cref="PipelinePolicy"/> used by a <see cref="ClientPipeline"/> to
+/// log request and response information.
+/// </summary>
+public class ClientLoggingPolicy : PipelinePolicy
+{
+    private const double RequestTooLongSeconds = 3.0; // sec
+
+    private readonly bool _logContent;
+    private readonly int _maxLength;
+    private readonly string _clientAssembly = "System-ClientModel";
+    private readonly ILogger _logger;
+    private readonly ClientModelLogMessages _logMessages;
```

Comment:
Curious: why are both _logger and _logMessages needed here, if _logMessages holds an ILogger property itself?